aioxmpp/avatar/service.py

```diff
@@ -509,19 +566,20 @@ def publish_avatar_set(self, avatar_set):
 
         id_ = avatar_set.png_id
 
-        yield from self._pubsub.publish(
-            None,
-            namespaces.xep0084_data,
-            avatar_xso.Data(avatar_set.image_bytes),
-            id_=id_
-        )
+        with (yield from self._lock):
```

Comment:
Nope. This will probably need a new pull request an possible additional support from PubSub (but I'll have to read the PubSub spec to be sure). As I do not want to download the image data to verify it matches, but simply check whether the image data pubsub item has the appropriate id.